File: desktop-app/src/main/java/entities/FanProfile.java
package entities;

public class FanProfile {
    private int id;
    private int userId;
    private String favoriteSport;
    private String country;
    private String bio;

    public FanProfile() {
    }

    public FanProfile(int userId, String favoriteSport, String country, String bio) {
        this.userId = userId;
        this.favoriteSport = favoriteSport;
        this.country = country;
        this.bio = bio;
    }

    public FanProfile(int id, int userId, String favoriteSport, String country, String bio) {
        this.id = id;
        this.userId = userId;
        this.favoriteSport = favoriteSport;
        this.country = country;
        this.bio = bio;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public String getFavoriteSport() {
        return favoriteSport;
    }

    public void setFavoriteSport(String favoriteSport) {
        this.favoriteSport = favoriteSport;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getBio() {
        return bio;
    }

    public void setBio(String bio) {
        this.bio = bio;
    }

    @Override
    public String toString() {
        return "FanProfile{id=" + id + ", userId=" + userId + ", favoriteSport='" + favoriteSport + "', country='"
                + country + "'}";
    }
}

File: smartfight/src/Entity/FanProfile.php
<?php

namespace App\Entity;

use App\Repository\FanProfileRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: FanProfileRepository::class)]
#[ORM\Table(name: 'fan_profile')]
class FanProfile
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\OneToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'user_id', nullable: false)]
    private User $user;

    #[ORM\Column(name: 'favorite_sport', type: 'string', length: 100, nullable: true)]
    private ?string $favoriteSport = null;

    #[ORM\Column(type: 'string', length: 100, nullable: true)]
    private ?string $country = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $bio = null;

    public function getId(): ?int { return $this->id; }
    public function getUser(): User { return $this->user; }
    public function setUser(User $user): static { $this->user = $user; return $this; }
    public function getFavoriteSport(): ?string { return $this->favoriteSport; }
    public function setFavoriteSport(?string $s): static { $this->favoriteSport = $s; return $this; }
    public function getCountry(): ?string { return $this->country; }
    public function setCountry(?string $c): static { $this->country = $c; return $this; }
    public function getBio(): ?string { return $this->bio; }
    public function setBio(?string $bio): static { $this->bio = $bio; return $this; }
}
